projects/06/hack_assembler/hack_assembler/consts.py

```diff
@@ -0,0 +1,117 @@
+import re
+from lark import Lark
+
+A_INSTRUCTION_MAX_LITERAL_SIZE = 32767
+
+LABEL_REGEX = re.compile('\((?P<label_name>.+)\)')
+
+JGT = "JGT"
+JEQ = "JEQ"
+JGE = "JGE"
+JLT = "JLT"
+JNE = "JNE"
+JLE = "JLE"
+JMP = "JMP"
+
+ZERO = "zero"
+ONE = "one"
+MINUS_ONE = "minus_one"
+DREGISTER = 'dregister'
+SREGISTER = 'sregister'
+NOT_DREGISTER = 'not_dregister'
+NOT_SREGISTER = 'not_sregister'
+MINUS_DREGISTER = 'minus_dregister'
+MINUS_SREGISTER = 'minus_sregister'
+DREGISTER_PLUS_ONE = 'dregister_plus_one'
+SREGISTER_PLUS_ONE = 'sregister_plus_one'
+DREGISTER_MINUS_ONE = 'dregister_minus_one'
+SREGISTER_MINUS_ONE = 'sregister_minus_one'
+DREGISTER_PLUS_SREGISTER = 'dregister_plus_sregister'
+DREGISTER_MINUS_SREGISTER = 'dregister_minus_sregister'
+SREGISTER_MINUS_DREGISTER = 'sregister_minus_dregister'
+DREGISTER_AND_SREGISTER = 'dregister_and_sregister'
+DREGISTER_OR_SREGISTER = 'dregister_or_sregister'
+
+COMP_TO_BINARY = {
+    ZERO: '101010',
+    ONE: '111111',
+    MINUS_ONE: '111010',
+    DREGISTER: '001100',
+    SREGISTER: '110000',
+    NOT_DREGISTER: '001101',
+    NOT_SREGISTER: '110001',
+    MINUS_DREGISTER: '001111',
+    MINUS_SREGISTER: '110011',
+    DREGISTER_PLUS_ONE: '011111',
+    SREGISTER_PLUS_ONE: '110111',
+    DREGISTER_MINUS_ONE: '001110',
+    SREGISTER_MINUS_ONE: '110010',
+    DREGISTER_PLUS_SREGISTER: '000010',
+    DREGISTER_MINUS_SREGISTER: '010011',
+    SREGISTER_MINUS_DREGISTER: '000111',
+    DREGISTER_AND_SREGISTER: '000000',
+    DREGISTER_OR_SREGISTER: '010101',
+}
+
+JMP_TO_BINARY = {
+    JGT: '001',
+    JEQ: '010',
+    JGE: '011',
+    JLT: '100',
+    JNE: '101',
+    JLE: '110',
+    JMP: '111',
+}
+
+lexer = Lark(f"""
+            instruction : A_PREFIX value | dest EQUAL comp | comp SEMICOLON jump

```

Comment:
Why did you assume that an assigment computation cannot contain a jump? 

e.g. 
A=D;JEQ 

is possible by the specification

Reply:
Actually I relied on the fact that assembling `Pong` worked when using our assembler.

Also, from the next exercise - it seems that this is not supported.

Reply:
Who cares?
it is actually easier to support it implementation wise, you go out of your way to implement assumptions you made up